packages/appium/lib/utils.js

```diff
@@ -46,6 +46,41 @@ function parseExtensionArgs (extensionArgs, extensionName) {
   return extensionSpecificArgs;
 }
 
+
+function parseKnownArgs (driverPluginArgs, argsConstraints) {
+  const knownArgNames = Object.keys(argsConstraints);
+  return _.toPairs(driverPluginArgs).reduce((args, [argName, argValue]) => {
+    if (knownArgNames.includes(argName)) {
+      args[argName] = argValue;
+    } else {
+      throw new Error(`"${argName}" is not a recognized key. Are you sure it's in the list ` +
+                      `of supported keys? ${JSON.stringify(knownArgNames)}`);
+    }
+    return args;
+  }, {});
+}
+
+/**
+ * Takes in a set of args, driver/plugin args passed in by user, and arg constraints
```

Comment:
The documentation must mention that args object is mutated there

Reply:
I'm actually in general not quite sure if it is a good idea to mutate the `args` property of the umbrella driver and whether this is not going to create unexpected conflicts for parallel sessions. @jlipps Could you please confirm it is safe?

Reply:
I think in general we want to move away from mutation unless there's a strong reason for it. We could avoid it pretty easily here by simply adding `args = _.cloneDeep(args);` as the first line of this method. then the caller can decide if it wants to mutate or not. but the place to mutate would be there and probably not in an incidental parser helper function.